arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -273,8 +273,7 @@ void arm_core_mpu_configure(u8_t type, u32_t base, u32_t size)
 	u32_t region_index = _get_region_index_by_type(type);
 	u32_t region_attr = _get_region_attr_by_type(type, size);
 
-	/* ARM MPU supports up to 16 Regions */
-	if (region_index > _get_num_regions()) {
```

Comment:
This looks like an out-of-bounds sanity-check bug, @agross-linaro, am I correct?